running-a-nats-service/nats_admin/jetstream_admin/encryption_at_rest.md

```diff
@@ -1,42 +1,66 @@
+
+Enabling encryption is done through the `jetstream` [configuration block](/running-a-nats-service/configuration/README.md#jetstream) on the server.
 
 ```text
 jetstream : {
-    key : “mykey”
-    store_dir: datastore
+  cipher: chachapoly
+  key : "mykey"
 }
 ```
 
-It is recommended to provide the encryption key through an environment variable, such as `JS_KEY` so it will not be persisted in a file.
+It is recommended to provide the encryption key through an environment variable at runtime, such as `$JS_KEY`, so it will not be persisted in a file.
 
 ```text
 jetstream : {
-    key: $JS_KEY
-    store_dir: datastore
+  cipher: chachapoly
+  key: $JS_KEY
 }
 ```
 
-You can pass this from the command line this way:
+The variable can be exported in the environment or passed when the server starts up.
 
 ```shell
-env JS_KEY="mykey" nats-server -c js.conf
+JS_KEY="mykey" nats-server -c js.conf
 ```
 
-We currently support two ciphers for encryption: ChaChaPoly20 and [XChaChaPoly1305](https://godoc.org/golang.org/x/crypto/chacha20poly1305). The default selected cipher depends on the platform.
+## Changing encryption settings
+
+### Enabling with existing data
+
+Enabling encryption on a server with existing data is supported. Do note that existing unencrypted message blocks will not be re-encrypted, however any new blocks that are stored _will_ be encrypted going forward.
 
-Note that message blocks are encrypted which includes message headers and payloads. Other metadata files are encrypted as well, such as the stream metadata file and consumer metadata files.
+If it is desired to encrypt the existing blocks, the stream can be backed up and restored (which decrypts on backup and then re-encrypts when restoring it).
 
-Starting a server with encryption enabled against a datastore that was not encrypted may result in failures when it comes to decrypting message blocks, which may not happen immediately upon startup. Instead, it will happen when attempting to deliver messages to consumers. However, when possible, the server will detect if the data was not encrypted and return the data without attempting to decrypt it.
 
-If the data is encrypted with a key and the server is restarted with a different key, the server will fail to decrypt messages when attempting to load them from the store. The server will still be active to support core NATS functionality. If this happens, you’ll see log messages like the following:
+### Disabling or changing the key
+
+If encryption was enabled on the server and the server is restarted with a different key or disabled all together, the server will fail to decrypt messages when attempting to load them from the store. If this happens, you’ll see log messages like the following:
 
 ```text
 Error decrypting our stream metafile: chacha20poly1305: message authentication failed
 ```
 
+Note, that this impact JetStream functionality, but the server will still support core NATS functionality.
```

Comment:
Are we missing 'will' in this sentence? 'Note, that this will impact JetStream...'